src/app/components/FrostedGlassPromo/index.tsx

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable jsx-a11y/anchor-has-content */
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { jsx } from '@emotion/react';
+import React, { PropsWithChildren, useContext } from 'react';
+import pick from 'ramda/src/pick';
+import Lazyload from 'react-lazyload';
+
+import IMAGE from '../Image';
+import makeRelativeUrlPath from '../../lib/utilities/makeRelativeUrlPath';
+import useClickTrackerHandler from '../../hooks/useClickTrackerHandler';
+import { RequestContext } from '../../contexts/RequestContext';
+
+import FrostedGlassPanel from './FrostedGlassPanel';
+import withData from './withData';
+
+import styles from './styles';
+
+const PANEL_OFFSET = 250;
+
+type FrostedGlassPromoProps = {
+  image: {
+    src: string;
+    srcset: string;
+    fallbackSrcset: string;
+    primaryMimeType: string;
+    fallbackMimeType: string;
+    sizes: string;
+    alt: string;
+    ratio: number;
+    width: number;
+    height: number;
+  };
+  footer: React.ReactNode | null;
+  url: string;
+  eventTrackingData: object | null;
+  index: number;
+  minimumContrast: number;
+  paletteSize: number;
+};
```

Comment:
Is it worth moving this to the the `types.ts` file?

Reply:
Its all good, the more eyes the better!

On this one I'm not so sure tbh. I quite like having the types as close to the thing they're type checking as possible. The main reason I put the others in the `types.ts` file was the re-use of some of them in tests. The `FormattedPromo` one could probably live in the `withData.tsx` file, as its not re-used anywhere else.

Its a hard one to decide on. Files/components can become quite noisy if there is a bunch of types flying around, so I can see both sides of the argument.